add details to start a node from a CometBFT snapshot (#47)
mezo-org/validator-kit: manual/README.md

    <br/><br/>

12. Apply to PoA. Please contact the Mezo team for further instructions.

> ## Starting a node from a CometBFT snapshot

Reviewer: Let's reorganize this a bit to make the structure of this document cleaner and more extendable for the future. I propose to do the following:
```
# Manual setup

## Prerequisites

## Setup 

## Runbooks 
// New parent section

### State sync from snapshot
// Add this description here

// Future ### if nedeed
```

Author: addressed in: 9dc4b0803ab5cde98258e9695ef0b95bcf6a7a60

> 

Reviewer: If we reorganize as proposed in the previous comment, let's link the 'Runbooks#State sync from snapshot' section from the root's `README.md`'s [same section](https://github.com/mezo-org/validator-kit?tab=readme-ov-file#state-sync-from-snapshot). I propose to replace:
```
Mezo team does not provide any snapshots yet. It will do so for Mezo Matsnet testnet in the near future. Until then, you can ask trusted community members for a snapshot.
```
with:
```
Mezo team provides snapshots for Mezo Matsnet testnet. Please refer to [this runbook](link to 'Runbooks#State sync from snapshot' from this document) for details. Alternatively, yoiu can ask trusted community members for a snapshot.
```

Author: addressed in: 9dc4b0803ab5cde98258e9695ef0b95bcf6a7a60

To start a node from a snapshot is no different to any other CometBFT based blockchain.

> To start with select a block from a trusted node, you can do so using the following API:
> `http://<NODE_OPERATOR_ADDRESS>:26657/block`, this will give you the very last block
> executed by the node.

Reviewer: ```suggestion
To start with, select a block from a trusted node. You can do so using the following API:
`http://<NODE_ADDRESS>:26657/block`, this will give you the very last block
executed by the node.
```

Then you will need to update the following fields in the cometbft config, you can find this
config under the following path `MEZOD_HOME/config/config.toml`:
- `enabled` should be set to true to enable state sync (to be disabled later, when
  the node have been running for normal stop and restart of the node without local state).
- `rpc_servers` is a list of trusted node rpc server that the light client can use
  to recovers blocks.
- `trust_height` should be the height of the selected trusted block
- `trust_hash`  should be the hash of the selected trusted block

> This is an example configuration you can use with the mezo matsnet, be sure to
> update the `trust_height` and `trust_hash` fields as explain previously everthing
> else would be valid as is:

Reviewer: ```suggestion
Here is an example configuration you can use for the Mezo Matsnet testnet. Be sure to
update the `trust_height` and `trust_hash` fields as explained previously. Everything
else would be valid as-is:
```

```toml
[statesync]
enable = true

rpc_servers = "mezo-node-0.test.mezo.org:26657,mezo-node-1.test.mezo.org:26657,mezo-node-2.test.mezo.org:26657,mezo-node-3.test.mezo.org:26657,mezo-node-4.test.mezo.org:26657,35.209.7.223:26657,35.209.252.61:26657"

trust_height = 1880001
trust_hash = "2185BC492BA0BD1FB1B0BFB16CE229736E850E5B64BB09B7363F9DB3EC5C2078"
trust_period = "112h0m0s"

discovery_time = "15s"

temp_dir = ""

chunk_request_timeout = "1m"

chunk_fetchers = "4"

> ```

Author: addressed in: 9dc4b0803ab5cde98258e9695ef0b95bcf6a7a60

> > NOTE: When selecting a block, be sure to have a block which is close to a snapshot.
> > Because the mezo matsnet takes snapshots every 5000 blocks, it is recommended to use
> > just the next block height (this help reduce timeout issues with the light client),
> > So if the last snapshot was on block 1885000, the best trusted block would be 1885001,
> > for which you can access the block hash with the following URL
> > `http://<NODE_OPERATOR_ADDRESS>:26657/block?height=188501`

Reviewer: ```suggestion
> [!NOTE]
> When selecting a block, be sure to have a block that is close to a snapshot.
> Matsnet nodes controlled by the Mezo team take snapshots every 5000 blocks. It is recommended to use
> the very next block height to reduce timeout issues that may occur on the bootstrapped node.
> For example, if the last snapshot was on block 1885000, the best trusted block would be 1885001,
> for which you can access the block hash with the following URL:
> `http://<NODE_ADDRESS>:26657/block?height=188501`
```

## Runbooks

### State sync from snapshot

To start a node from a snapshot is no different to any other CometBFT based blockchain.

To start with, select a block from a trusted node. You can do so using the following API:
`http://<NODE_ADDRESS>:26657/block`, this will give you the very last block
executed by the node.

Once done, you will need to update the following fields in the CometBFT config. You can find this
config in the `MEZOD_HOME/config/config.toml` file, under the `statesync` TOML section:
- `enable`: should be set to `true` to enable state sync (can be flipped back once the node synchronizes)
- `rpc_servers`: is a list of trusted node RPC servers that the node can use to recover blocks from
- `trust_height`: should be the height of the selected trusted block
- `trust_hash`:  should be the hash of the selected trusted block

Moreover, please make sure your node doesn't have any local state while using state sync.

Here is an example configuration you can use for the Mezo Matsnet testnet. Be sure to
update the `trust_height` and `trust_hash` fields as explained previously. Everything
else would be valid as-is:

> ```toml

Reviewer: Let's fix formatting of this code block:
<img width="1049" alt="image" src="https://github.com/user-attachments/assets/30b29b6a-a19f-451a-b608-80436e1ccc21" />
